components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/MessageProcessorResource.java

```diff
@@ -0,0 +1,252 @@
+
+    //Message processor name query parameter
+    private static final String PROCESSOR_NAME_PARAM = "name";
+    //Message processor types
+    private static final String SAMPLING_PROCESSOR_TYPE = "Sampling-processor";
+    private static final String SCHEDULED_MESSAGE_FORWARDING_TYPE = "Scheduled-message-forwarding-processor";
+    private static final String FAILOVER_SCHEDULED_MESSAGE_FORWARDING_TYPE = "Failover-scheduled-message-forwarding-processor";
+    private static final String CUSTOM_PROCESSOR_TYPE = "Custom-message-processor";
+    //Message processor statuses
+    private static final String PROCESSOR_ACTIVE_STATUS = "active";
+    private static final String PROCESSOR_INACTIVE_STATUS = "inactive";
+    //Constants for message-processor JSON object
+    private static final String TYPE_PROPERTY = "type";
+    private static final String CONTAINER_PROPERTY = "artifactContainer";
+    private static final String FILE_NAME_PROPERTY = "fileName";
+    private static final String PARAMETER_PROPERTY = "parameters";
+    private static final String MESSAGE_STORE_PROPERTY = "messageStore";
+    private static final String STATUS_PROPERTY = "status";
+
+    public MessageProcessorResource(String urlTemplate) {
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        Set<String> methods = new HashSet<>();
+        methods.add("GET");
+        methods.add("POST");
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext) {
+        //Building message since POST requests are facilitated
+        buildMessage(messageContext);
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+
+        if (messageContext.isDoingGET()) {
+            String nameParameter = Utils.getQueryParameter(messageContext, PROCESSOR_NAME_PARAM);
+            if (Objects.nonNull(nameParameter)) {
+                populateMessageProcessorData(messageContext, nameParameter);
+            } else {
+                populateMessageProcessorList(messageContext);
+            }
+            axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        } else if (Utils.isDoingPOST(axis2MessageContext)) {
+            try {
+                if (!JsonUtil.hasAJsonPayload(axis2MessageContext)) {
+                    return false;
+                }
+                JsonObject payload = getJsonPayload(messageContext);
+                if (payload.has(PROCESSOR_NAME_PARAM) && payload.has(STATUS_PROPERTY)) {
+                     changeProcessorStatus(messageContext, payload);
+                }
+            } catch (IOException e) {
+                LOG.error("Error when parsing JSON payload", e);
+                Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Error when parsing JSON payload"));
+            }
+        }
```

Comment:
Shall we send an error message if it is not a GET or a POST?

Reply:
I suppose this will be handled at the Internal Api level [1] & [2]  since we are registering only GET and POST and the call will not propagate here. Do we need to handle it again here

[1] - https://github.com/wso2/carbon-mediation/blob/master/components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/InternalAPIDispatcher.java#L62
[2] - https://github.com/wso2/carbon-mediation/blob/master/components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/InternalAPIDispatcher.java#L95

Reply:
This resource will inly be called if the request is either GET or POST